### PR TITLE
Update References and Templates concept page.

### PR DESCRIPTION
**Pull request by StephanieJKS**
Closes ~#ISSUE_NUMBER~

### What are the changes and their implications?
Makes changes to the References and Templates concept page. These changes are made to generally improve the page, and make it more relevant to v4.

## Checklist

- [x] Pull request is made to the "develop" branch
- [ ] Tests added
- [x] Documentation added/updated
- [x] Code has been formatted with Prettier
- [x] Edits from maintainers are allowed

### CODE REVIEW

File: packages/docs/concepts/references-and-templates.yaml [hunk 1]
```diff
@@ -81,7 +85,7 @@ _ref:
 
             ## Nunjucks Templating
 
-            Templating can be taken further by referencing [Nunjucks](https://mozilla.github.io/nunjucks/) template files. If a file ends with the `.njk` file extension, the file will first be hydrated as a Nunjucks template, using the `vars` as template variables. If the file ends with `.yaml.njk` or `.json.njk`, the output of the template will then be parsed. Nunjucks templates are useful since the template file does not need to be valid yaml before it is hydrated, and features like for-loops and if-statements can be used.
+            Templating can be taken further by referencing [Nunjucks](https://mozilla.github.io/nunjucks/) template files by simply adding .njk to the file extensions, for example, .yaml.njk. Nunjucks templates hydrate variables at build time, before the yaml is parsed, enabling Nunjucks features like for-loops and if-statements.
```
Comment (Gervwyk) [inline packages/docs/concepts/references-and-templates.yaml:88]: I think I prefer the original paragraph here, it feels like some details are lost here. Not sure, @SamTolmay, @JohannMoller  what do you guys think?

Comment (SamTolmay) [reply packages/docs/concepts/references-and-templates.yaml:93]: I agree, let's stick with the previous version

Comment (JohannMoller) [reply packages/docs/concepts/references-and-templates.yaml:93]: Ja same. The previous one was a bit more descriptive

File: packages/docs/concepts/references-and-templates.yaml [hunk 2]
```diff
@@ -24,7 +24,14 @@ _ref:
         type: MarkdownWithCode
         properties:
           content: |
-            References are used to split the configuration of an app into logically distinct files, and to reuse configuration in the app. References can be used almost anywhere in the configuration, as long as the configuration remains valid YAML.
+            References and templates have the following use cases:
+            - Splitting out the config for readability and code navigation.
+            - Splitting out a piece of config in order to use the exact same config in multiple places.
+            - Using a reference with variables to make use of a shared piece of config with dynamic variables.
+            - Using an njk file which allows for much more complicated things - this is an advanced feature and easily overused.
```
Comment (SamTolmay) [inline packages/docs/concepts/references-and-templates.yaml:31]: Lets say something like:
Using a `.njk` file to create config files using the Nunjucks templating language.
Using resolver and transformer functions to create Lowdefy config using Javascript functions.

File: packages/docs/concepts/references-and-templates.yaml (hunk 1 above)
Comment (StephanieJKS) [reply packages/docs/concepts/references-and-templates.yaml]: Change made in d9ad6e012a786eeb90ec2e938dd465040c158fb1

File: packages/docs/concepts/references-and-templates.yaml (hunk 2 above)
Comment (StephanieJKS) [reply packages/docs/concepts/references-and-templates.yaml]: Change made in d9ad6e012a786eeb90ec2e938dd465040c158fb1

**Review by SamTolmay — APPROVED** [review]

**Pull request merged**